### ISSUE TITLE
Support transparent proxy cache

### ISSUE THREAD

**Issue by steven-zou** [szlabs/harbor-automation-4k8s#10]
Migrate work delivered here: https://github.com/indeedeng-alpha/harbor-container-webhook to this project

**Comment by cnmcavoy** [comment]
I took a brief dive into the codebase of the project, and I found that you have implemented per-namespace image rewriting. The namespaced image rewriting relies on the HarborServerConfiguration CRD, and both that CRD and namespace annotations must be set to enable image mutation. Does that summary seem correct?

Assuming I understand that correctly, I think the operator is missing cluster-scoped image rewriting. I think we could add an optional field `ImageRules` to the HarborServerConfiguration to map registries to harbor projects. Here is an example of what a sample of the CRD might look like:
```
---
apiVersion: goharbor.goharbor.io/v1alpha1
kind: HarborServerConfiguration
metadata:
  name: harborserverconfiguration-sample
spec:
  serverURL: goharbor.io
  accessCredential:
    namespace: kube-system
    accessSecretRef: mysecret
  version: 2.1.0
  http:
    timeout: 20s
    scheme: https
    skipTLSVerify: false
  imageRules:
    rules: # a list of registries and the harbor projects that contain a cache
      - registry: docker.io # docker.io is a special case, any "bare" docker reference will also be rewritten to use this harbor project
        project: dockerhub-cache
      - registry: quay.io
        project: quay-cache
    healthcheck: true # whether the controller checks if harbor is up before rewriting the container images
``` 

The namespace annotations would supersede these cluster rules, if they are present.

**Comment by cnmcavoy** [comment]
Solved by #12
